src/models/quote.js

```diff
@@ -1,38 +1,29 @@
 'use strict'
 
-const request = require('co-request')
 const UnacceptableExpiryError = require('../errors/unacceptable-expiry-error')
 const UnacceptableAmountError = require('../errors/unacceptable-amount-error')
 const AssetsNotTradedError = require('../errors/assets-not-traded-error')
-const ExternalError = require('../errors/external-error')
 const DEFAULT_DESTINATION_EXPIRY = 5 // seconds
 
 function * makeQuoteQuery (params) {
-  const sourceLedger = params.source_ledger ||
-    (yield getAccountLedger(params.source_account))
-  const destinationLedger = params.destination_ledger ||
-    (yield getAccountLedger(params.destination_account))
+  const sourceLedger = addressToLedger(params.source_address)
+  const destinationLedger = addressToLedger(params.destination_address)
   const explain = params.explain === 'true'
   return {
     sourceLedger,
     destinationLedger,
     sourceAmount: params.source_amount,
     destinationAmount: params.destination_amount,
+    destinationPrecisionAndScale: params.destination_precision && {
+      precision: params.destination_precision,
+      scale: params.destination_scale
+    },
     explain
   }
 }
 
-function * getAccountLedger (account) {
-  const res = yield request({
-    method: 'get',
-    uri: account,
-    json: true
-  })
-  const ledger = res.body && res.body.ledger
-  if (res.statusCode !== 200 || !ledger) {
-    throw new ExternalError('Unable to identify ledger from account: ' + account)
-  }
-  return ledger
+function addressToLedger (address) {
+  return address.split('.').slice(0, -1).join('.')
```

Comment:
> But in this case I imagine there is no third party market maker right?

That's correct.

>  So if the us.fed.wf.stefan account balance goes down, how does that balance out on the us.fed.wf ledger?

The balance for `us.fed.wf.stefan` wouldn't go down. It would go *up*.

Somebody will prepare a transfer to me (as `us.fed.wf.stefan`) on the `us.fed.wf.` ledger. I (as the connector to my subledger) prepare a transfer out of the transactional account on the `us.fed.wf.stefan.` ledger to `us.fed.wf.stefan.phone`.

In plain English - if you send money to my phone, my bank account balance will be higher afterwards. If my phone spends money, my bank account balance will be lower.